pages/docs/manual/latest/polymorphic-variant.mdx

```diff
@@ -0,0 +1,273 @@
+---
+title: "Polymorphic Variant"
+description: "The Polymorphic Variant data structure in ReScript"
+canonical: "/docs/manual/latest/polymorphic-variant"
+---
+
+# Polymorphic Variant
+
+Now that we know what [variant types](./variant) are, let's dive into a more specific version, so called polymorphic variants (or poly variants).
+
+First off, here are some key features:
+
+- Poly variants are **structurally typed** (in comparison to **nominally typed** variants). They can be used without an explicit type definition.
+- They allow easier JavaScript interop (compile to strings / objects with predictable `NAME` and `VAL` attribute) and don't need explicit runtime conversions, unlike common variants.
+- Due their structural nature, they oftentimes cause tricky type checking errors when types don't match up, which makes them a more advanced feature.
```

Comment:
```suggestion
- Due to their structural nature, they oftentimes cause tricky type checking errors when types don't match up, which makes them a more advanced feature.
```

Reply:
it is the combination of `structural + recursive types ` which makes the error message confusing

Reply:
I mentioned that somewhere in the variant / poly variant comparison in the end. Couldn't replicate any example where an error message gets completely out of hand, unless you are using very thorough libraries such as Tyxml